python/tank/pipelineconfig_factory.py

```diff
@@ -279,29 +278,38 @@ def _get_configuration_context():
         # Once found, we can double check that the current Entity is actually
         # associated with the project that the PC is associated with.
         val = pipelineconfig_utils.get_config_install_location(curr_pc_path)
-        
+
     return val
 
 
+def _cleanup_separators(path):
+    """
+    Flips all slashes to the operating system's standard slash separator and strips
```

Comment:
Actually, I've poked around, and I've stumbled unto this: https://github.com/shotgunsoftware/tk-core/blob/8f8c04a71ad7607fecb29713e553b2b8ea40f68f/python/tank/pipelineconfig_utils.py#L136
How about we make that method public?

That's even better than what I have, since it takes care of double slashes. I'll reuse that instead. Plus it handles the C:\ scenario correctly.

This brings to mind something I've been thinking for a while: we should introduce a base module that is meant to be use for common code patterns like normalizing paths, exception safe pushing and popping the state of something (like the umask) using the with statement and other such reusable components. Right now we are at the mercy of reusing code across modules that maybe don't  have anything related to each other.